core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -280,6 +280,8 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
 
   private[spark] def ui: Option[SparkUI] = _ui
 
+  def uiWebUrl: Option[String] = _ui.map(_.webUrl)
```

Comment:
I understand now why it's done this way. Yes I forgot that this is of course executed in the shell which doesn't have access to private[spark] stuff. Hm, I like the change. I guess I'm a little wary of tacking on yet another method in SparkContext just for it. I wonder if anyone else has any particular feeling o this? a pointer to the UI does sound handy.